.github/workflows/cd-publishRelease.yml

```diff
@@ -0,0 +1,88 @@
+name: Publish Release
+
+on:
+    workflow_dispatch:
+      inputs:
+        target-branch:
+          description: 'Branch to target'
+          required: true
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+
+    steps:
+     - name: Checkout code
+       uses: actions/checkout@v3
+       with:
+        ref: ${{ github.event.inputs.target-branch }}
+        token: ${{ secrets.TAG_TOKEN }}
+
+     - name: Run ballerina build
+       uses: ballerina-platform/ballerina-action@2201.7.0
+       with:
+        args:
+          build
+       env:
+          WORKING_DIR: ${{ github.workspace }}
+          JAVA_HOME: /usr/lib/jvm/default-jvm   
+
+     - name: Find JAR file
+       id: find-jars
+       run: |
+          jar_files=$(find "${{ github.workspace }}" -name "*.jar" -path "*/target/bin/*")
+          echo "JAR_FILES=$jar_files" >> $GITHUB_ENV
+
+     - name: JAR file Info
+       run: |
+          echo "Jar path: $JAR_FILES"
+          jar tf $JAR_FILES
+
+     - name: Get tag from Ballerina.toml and create tag name
```

Comment:
We don't have to create a tag explicitly and push. Creating a release will create a tag.